### DISCUSSION TITLE
How use in electron + better-sqlite3

### DISCUSSION THREAD

**Discussion by aquarius-wing** [vlcn-io/cr-sqlite#424] [Q&A] [answered]
Hello everyone, first of all, thank you for developing this cool application.

I really want to use it in my own project, https://meogic.com. It is a project with a tech stack that uses electron as the client tool, knex for SQL encapsulation and migration, and objectionjs for ORM.

I tried reading the official documentation at https://vlcn.io/docs/cr-sqlite/js/nodejs, then looked at https://vlcn.io/docs/cr-sqlite/js/reactivity, which led me to the following code:

```javascript
import SQLiteDB from 'better-sqlite3'
import { extensionPath } from "@vlcn.io/crsqlite";

const db = new SQLiteDB(pathToDb)
db.loadExtension(extensionPath)
const rx = tblrx(db)

const disposable = rx.onRange(["posts"], (updateTypes) => {
  console.log("posts changed", updateTypes);
});
```

Then I got an error, 

> TypeError: this.db.onUpdate is not a function
    at new TblRx (file:///Users/wing/Develop/electron-vite-vue/node_modules/@vlcn.io/rx-tbl/dist/tblrx.js:33:37)
    at tblrx (file:///Users/wing/Develop/electron-vite-vue/node_modules/@vlcn.io/rx-tbl/dist/tblrx.js:164:12)
    at file:///Users/wing/Develop/electron-vite-vue/dist-electron/main/index.js:12:12
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

I have no idea how to handle it, please help me.

Thanks!

**Comment by tantaman** [comment, marked as answer]
Thanks for reporting. `better-sqlite3` doesn't expose the sqlite update hook :(

We could craft a subscription implementation that uses triggers (as suggested here: https://github.com/WiseLibs/better-sqlite3/issues/62) or expose the update_hook in better-sqlite3 and hope they take the PR.

**Reply by GitMurf** [reply]
@tantaman we have same use case. Using electron with better SQLite3. Was there any progress on this? Is there a solution to use the trigger as Joshua mentioned from better sqlite3? Thanks!
